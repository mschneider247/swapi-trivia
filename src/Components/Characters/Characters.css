.charac-card {
  background-color: #535353;
  display: flex;
  flex-direction: column;
  margin: 20px;
  padding: 10px 20px 20px 20px;
  text-align: left;
  width: 250px;
}

h3 {
  color: #000000;
  font-size: 2rem;
  margin: 0 0 10px 0;
}

h4 {
  font-weight: lighter;
  margin: 5px;
}

h5 {
  color: #000000;
}

.bold-text {
  font-weight: bold;
}

.view-charac {
  background-color: #EEDB00;
  display: flex;
  justify-content: center;
  margin: 0 20px 20px 20px;
}

.favorite {
  background-image: url('../../images/swapi_heart_unclicked.svg');
  background-repeat: no-repeat;
  height: 40px;
}
