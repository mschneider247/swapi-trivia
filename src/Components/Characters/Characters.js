import React from 'react'
import { Component } from 'react';
import './Characters.css'
import { isProperty } from '@babel/types';

class Characters extends Component {
  constructor(){
    super()
    this.state = {
      isFavorited: false
    }
  }

  render() {
    return (
      <section className="charac-card">
        <div className="favorite" onClick={() => {
          this.props.updateFavoriteCharacters(this.props)
          this.state.isFavorited ? this.setState({ isFavorited: false}) : this.setState({ isFavorited: true})
          }}>
        </div>
        <h3>{this.props.name}</h3>
        <h4><span className="bold-text">SPECIES </span>{this.props.species}</h4>
        <h4><span className="bold-text">HOMEWORLD </span>{this.props.homeworld}</h4>
        <h4><span className="bold-text">POPULATION </span>{this.props.population}</h4>
        <h4><span className="bold-text">OTHER MOVIES </span>{this.props.films}</h4>
      </section>
    )
  }
}

export default Characters;
