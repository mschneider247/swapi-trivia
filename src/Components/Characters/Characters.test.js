import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('Characters', () => {

  it('should match snapshot with proper data', () => {
    const mockCheckFavoriteStatus = jest.fn()
    const mockUpdateFavoriteCharacters = jest.fn()
    const wrapper = shallow(<
      Characters
        key={1}
        id={1}
        name="Mike"
        homeworld="Turing"
        species="Boss"
        films="Life In A Basement"
        population={30}
        updateFavoriteCharacters={mockUpdateFavoriteCharacters}
        url="www.turing.io"
        isFavorite={true}
        checkFavoriteStatus={mockCheckFavoriteStatus}
      />)

      expect(wrapper).toMatchSnapshot();
  })

  it('should call updateFavoriteCharacters on click', () => {
    const mockUpdateFavoriteCharacters = jest.fn()
    const mockCheckFavoriteStatus = jest.fn()
    const wrapper = shallow(<
      Characters
        key={1}
        id={1}
        name="Mike"
        homeworld="Turing"
        species="Boss"
        films="Life In A Basement"
        population={30}
        updateFavoriteCharacters={mockUpdateFavoriteCharacters}
        url="www.turing.io"
        isFavorite={true}
        checkFavoriteStatus={mockCheckFavoriteStatus}
      />)

    wrapper.find('div').simulate('click')

    expect(mockUpdateFavoriteCharacters).toHaveBeenCalled();

  })

})
