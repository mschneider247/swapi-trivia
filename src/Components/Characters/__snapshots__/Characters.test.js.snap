// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters should match snapshot with proper data 1`] = `
<section
  className="charac-card"
>
  <div
    className="unfavorite"
    onClick={[Function]}
  />
  <h3>
    Mike
  </h3>
  <h4>
    <span
      className="bold-text"
    >
      SPECIES 
    </span>
    Boss
  </h4>
  <h4>
    <span
      className="bold-text"
    >
      HOMEWORLD 
    </span>
    Turing
  </h4>
  <h4>
    <span
      className="bold-text"
    >
      POPULATION 
    </span>
    30
  </h4>
  <h4>
    <span
      className="bold-text"
    >
      OTHER MOVIES 
    </span>
    Life In A Basement
  </h4>
</section>
`;
